Libraries/CustomComponents/ListView/ListView.js

```diff
@@ -368,14 +383,30 @@ var ListView = React.createClass({
             rowID,
             adjacentRowHighlighted
           );
-          bodyComponents.push(separator);
-          totalIndex++;
+          sectionComponents.push(separator);
         }
+        countSectionLines++;
         if (++rowCount === this.state.curRenderedRowsCount) {
           break;
         }
       }
-      if (rowCount >= this.state.curRenderedRowsCount) {
+      var isEarlyBreak = rowCount >= this.state.curRenderedRowsCount;
+      if (isEarlyBreak || countSectionLines === rowIDs.length) {
+        //early break or end section
+        var style = this.props.sectionStyle;
+        bodyComponents.push(
+          React.cloneElement(
+            <View />,
+            {
+              key: 'section_' + sectionID,
+              style: typeof style == 'function' ? style(sectionID) : style
+            },
+            sectionComponents
```

Comment:
This is going to wrap every row in an extra unnecessary view, even if people don't want to use this new sectionStyle feature, which will hurt perf and probably break apps.  If we decide we want this as a feature, we'll need to make sure it doesn't change behavior when it's not used.